src/portable/st/stm32_fsdev/dcd_stm32_fsdev_pvt_st.h

```diff
@@ -119,26 +119,50 @@
 // Volatile is also needed to prevent the optimizer from changing access to 32-bit (as 32-bit access is forbidden)
 static __IO uint16_t * const pma = (__IO uint16_t*)USB_PMAADDR;
 
+// This array maps the endpoint address to a endpoint register in hardware
+// By default the tinyusb stack chooses the endpoint register using the endpoint number
+// However when using ISOCHRONOUS endpoints, the entire endpoint register (RX _and_ TX)
+// are used for a double buffer in a single direction. If you now want to use the same endpoint number with
+// different directions (e.g. 0x02 and 0x82), they would be mapped to the same register
+// In this case, this serves as a crude helper to overcome this limitation by implementing
+// this callback and defining a custom mapping in your application
+TU_ATTR_WEAK uint8_t tu_stm32_edpt_number_cb(uint8_t addr);
+
+// This callback is called on SOF and can be used to e.g. capture a timer value for timing purposes
+TU_ATTR_WEAK void tu_stm32_sof_cb(void);
```

Comment:
@skuep 

For the moment I kept your SOF callback but for the stack it's not quite the right place to do it.

Did you test the `tud_audio_feedback_interval_isr` and found out it has too much jitter ?

If a prioritized SOF callback is needed we need do it in a generalized way for other DCD drivers.

Reply:
Yeah I am also still on the fence with this callback. 
From the looks of it, the generated feedback values in the speaker feedback endpoint had way less jitter for the UAC2 application. However I did not check if this actually makes a difference for the (stability of the) feedback control loop running on the Host. Since I can only test Linux hosts, I left it in there for a better gut feel in terms of compatibility.

I do however think it would be a nice addition, if this would be a more general solution across all DCD drivers.